magma/conversions.py

```diff
@@ -193,12 +193,12 @@ def zext(value, n):
         zeros = sint(0, n)
     elif isinstance(value, BitsType):
         zeros = bits(0, n)
-    return concat(zeros, value)
+    return concat(value, zeros)
 
 
 def sext(value, n):
     assert isinstance(value, SIntType)
-    return sint(concat(array(value[-1], n), array(value)))
+    return sint(concat(array(value), array(value[-1], n)))
```

Comment:
@phanrahan can you review these changes? I think they look okay to me, but would be good to get a second set of eyes since these bit related ops can be tricky. From my understanding, `value[-1]` (so the last element in the array of bits) contains the sign bit, so to properly extend it, we need to concat the sign value on the end of the array. The previous implementation was concatenating it to the front of the array. This assumes that the concat logic matches the indexing logic, where the end values of concat are actually prepended (if you consider the most significant bits being in the front).

There's some wonkiness here, we'll want to review that the indexing logic and the concatenation logic are consistent